core/services/workflows/syncer/workflow_registry.go

```diff
@@ -452,93 +497,84 @@ func (w *workflowRegistry) handleWorkflowMetadataAsEvents(ctx context.Context, w
 			engineKeyMap[wfMeta.WorkflowName+hex.EncodeToString(wfMeta.Owner)] = true
 			continue
 		}
 
-		// if the workflow engine is running, the metadata is the same, but the status isn't active
-		// then the workflow is has been paused. NOOP to be handled below as a deleted event, which clears the DB workflow spec.
+		// if the workflow engine is in the registry, the metadata is the same, but the status isn't active
+		// then the workflow has been paused. NOOP - to be handled below as a deleted event, which clears the DB workflow spec.
 	}
 
 	// Shut down engines that are no longer in the contract's latest workflow metadata state
 	allEngines := w.engineRegistry.GetAll()
 	for _, engine := range allEngines {
-		keyStr := engine.workflowName + hex.EncodeToString(engine.workflowOwner)
-		_, exists := engineKeyMap[keyStr]
-
+		prevWfID := hex.EncodeToString(engine.workflowID[:])
+		_, exists := engineKeyMap[prevWfID]
 		if !exists {
 			toDeletedEvent := WorkflowRegistryWorkflowDeletedV1{
 				WorkflowID:    engine.workflowID,
 				WorkflowOwner: engine.workflowOwner,
 				DonID:         donID,
 				WorkflowName:  engine.workflowName,
 			}
-			err := w.handler.Handle(ctx, workflowAsEvent{
+			events = append(events, workflowAsEvent{
 				Data:      toDeletedEvent,
 				EventType: WorkflowDeletedEvent,
 			})
-			if err != nil {
-				w.lggr.Errorw("failed to handle event", "err", err, "type", WorkflowDeletedEvent)
-			}
 		}
 	}
-}
 
-// pollRegistryState polls every w.ticker tick for the latest workflow state and
-// calls the handler to bring the system's state up to date with the difference.
-func (w *workflowRegistry) pollRegistryStateToEvents(ctx context.Context, don capabilities.DON, reader ContractReader) {
-	ticker := w.getTicker()
-
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		case <-ticker:
-			workflowMetadata, _, err := w.getWorkflowMetadata(ctx, don, reader)
-			if err != nil {
-				w.lggr.Errorw("failed to process registry state", "err", err)
-				continue
-			}
-			w.handleWorkflowMetadataAsEvents(ctx, workflowMetadata, don.ID)
-		}
+	for _, event := range events {
+		w.eventCh <- event
 	}
 }
 
 // syncUsingReconciliationStrategy syncs workflow registry contract state by polling the workflow metadata state and comparing to local state.
 // It still watches for ForceUpdateSecretsEvents, which can't be reconciled through workflow metadata state.
 func (w *workflowRegistry) syncUsingReconciliationStrategy(ctx context.Context, don capabilities.DON, reader ContractReader) {
 	workflowMetadata, loadWorkflowsHead, err := w.getWorkflowMetadata(ctx, don, reader)
-	// TODO: if this fails, we should still start the event looping
 	if err != nil {
 		w.lggr.Errorw("failed to load initial workflows", "err", err)
-		return
 	}
-	w.handleWorkflowMetadataAsEvents(ctx, workflowMetadata, don.ID)
-
-	w.pollRegistryStateToEvents(ctx, don, reader)
+	w.workflowMetadataToEvents(ctx, workflowMetadata, don.ID)
```

Comment:
You don't need the first synchronous tick here; you can just spin up both of the goroutines below and you'll be done

From an ordering perspective it's fine; if you get a secrets event before a workflow event it should all work as expected

Reply:
Since I already need to run the query to get the head I figured it made sense, but I agree that it's not _neccessary_. I'll remove it.